pages/api/index.ts

```diff
@@ -71,11 +88,19 @@ export default async (req: NextApiRequest, res: NextApiResponse): Promise<void>
 
         // Get recently played tracks
         const username = await getUsername(tokens.accessToken);
-        const recentlyPlayed = await getRecentlyPlayed(count, tokens.accessToken);
+        const limit = uniqueTrack ? Constants.defaultUniqueTrackSearchLimit : count
+        let recentlyPlayedItems = await getRecentlyPlayed(limit, tokens.accessToken);
+
+        if (uniqueTrack) {
+            // Remove duplicate tracks...
+            recentlyPlayedItems = recentlyPlayedItems.filter((v, i, a) => a.findIndex((t) => t.track.id === v.track.id) === i);
+            // ... then only return the number of items that are requested.
+            recentlyPlayedItems.slice(0, count);
```

Comment:
`slice()` returns a copy and doesn't modify in-place, so this should be `recentlyPlayedItems = ...`,

Reply:
Do you mean` recentlyPlayedItems = [...recentlyPlayedItems.slice(0, count)];`? Not familiar with JS :)

Reply:
`recentlyPlayedItems = recentlyPlayedItems.slice(0, count);` should work

Reply:
@JeffreyCA I pushed some updates :)

Reply:
Thanks!